src/briefcase/platforms/linux/flatpak.py

```diff
@@ -1 +1,200 @@
+    description = "Create and populate a Linux Flatpak."
+
+    @property
+    def support_package_url(self):
+        """The URL of the support package to use for apps of this type.
+
+        Flatpak uses the original CPython sources, and compiles them in
+        the flatpak sandbox.
+        """
+        base_version = ".".join(str(m) for m in self.sys.version_info[:3])
+        full_version = self.stdlib_platform.python_version()
+        return f"https://www.python.org/ftp/python/{base_version}/Python-{full_version}.tgz"
+
+    def output_format_template_context(self, app: AppConfig):
+        """Add flatpak runtime/SDK details to the app template."""
+        return {
+            "flatpak_runtime": self.flatpak_runtime(app),
+            "flatpak_runtime_version": self.flatpak_runtime_version(app),
+            "flatpak_sdk": self.flatpak_sdk(app),
+        }
+
+    def _unpack_support_package(self, app, support_file_path):
+        """Unpack the support package.
+
+        Flatpak doesn't unpack the support package; it copies the
+        tarball as-is into the source tree.
+        """
+        with self.input.wait_bar("Installing support file ..."):
+            self.shutil.copy(
+                support_file_path,
+                self.bundle_path(app) / support_file_path.name,
+            )
+
+    def install_app_dependencies(self, app: AppConfig):
+        """Configure application dependencies.
+
+        Flatpak needs to install it's own dependencies as part of the
+        build process, so we override the base implementation to write a
+        requirements file.
+        """
+        with self.input.wait_bar("Writing dependency file..."):
+            with (self.bundle_path(app) / "requirements.txt").open("w") as f:
+                if app.requires:
+                    for requirement in app.requires:
+                        f.write(f"{requirement}\n")
+
+
+class LinuxFlatpakUpdateCommand(LinuxFlatpakMixin, UpdateCommand):
+    description = "Update an existing Linux Flatpak."
+
+
+class LinuxFlatpakBuildCommand(LinuxFlatpakMixin, BuildCommand):
+    description = "Build a Linux Flatpak."
+
+    def build_app(self, app: AppConfig, **kwargs):
+        """Build an application.
+
+        :param app: The application to build
+        """
+        self.logger.info(
+            "Ensuring Flatpak runtime for the app is available...",
+            prefix=app.app_name,
+        )
+        flatpak_repo_alias, flatpak_repo_url = self.flatpak_runtime_repo(app)
+
+        with self.input.wait_bar("Ensuring Flatpak runtime repo is registered..."):
+            self.flatpak.verify_repo(
+                repo_alias=flatpak_repo_alias,
+                url=flatpak_repo_url,
+            )
+
+        # ``flatpak install`` uses a lot of console animations, and there
+        # doesn't appear to be a way to turn off these animations. Use those
+        # native animations rather than wrapping in a wait_bar.
```

Comment:
furthermore, this actually brought up a user prompt for me to clarify which SDK package i wanted to install when I used `org.gnome.SDK`....not sure input would work with a wait bar.

Reply:
Interesting... I didn't get that prompt; what was your project configuration to get that prompt? I got an error if I specified:
```
flatpak_runtime = "org.gnome.Platform"
flatpak_sdk = "org.gnome.Sdk"
```
(because there isn't a gnome version 21.08 runtime); if I add `flatpak_runtime_version = "42"`, it installed without a prompt. Did you use something different?

Reply:
hmmm....I used the same config. This is also a fairly freshly installed VM too....so I don't think I've added anything crazy to flatpak.

```
[tool.briefcase.app.flat.linux]
template = "https://github.com/beeware/briefcase-linux-flatpak-template"
template_branch = "dev"
flatpak_runtime = "org.gnome.Platform"
flatpak_runtime_version = "42"
flatpak_sdk = "org.gnome.SDK"
```

```
❯ briefcase build linux flatpak

[flat] Ensuring Flatpak runtime for the app is available...
Ensuring Flatpak runtime repo is registered... done
Looking for matches…
Skipping: org.gnome.Platform/x86_64/42 is already installed
Similar refs found for ‘org.gnome.SDK’ in remote ‘flathub’ (user):

   1) runtime/org.gnome.Sdk.Compat.i386/x86_64/42
   2) runtime/org.gnome.Sdk.Compat.aarch64/x86_64/42
   3) runtime/org.gnome.Sdk/x86_64/42
   4) runtime/org.gnome.Sdk.Docs/x86_64/42

Which do you want to use (0 to abort)? [0-4]: 0
error: No ref chosen to resolve matches for ‘org.gnome.SDK’

Unable to install Flatpak runtime org.gnome.Platform//42 and SDK org.gnome.SDK//42 from repo flathub.

Log saved to /home/user/tmp/beeware/flat/briefcase.2022_07_26-21_47_37.build.log
```

[briefcase.2022_07_26-21_47_37.build.log](https://github.com/beeware/briefcase/files/9194384/briefcase.2022_07_26-21_47_37.build.log)

Reply:
And, to top it off, it asks me every time I build the flatpak which `org.gnome.SDK` I want to install....:/

Reply:
Hrm... I wonder if it might be related to the work you've been doing with i386. If the existence of 2 Platform flatpaks means there's 2 that match `org.gnome.SDK//42`, perhaps specifying `org.gnome.SDK/x86_64/42` will remove the ambiguity...

Reply:
@rmartin16 I've just pushed an update that will be explicit about architecture when installing the SDK; does that work better for you?

Reply:
same :/

```
[flat] Ensuring Flatpak runtime for the app is available...

>>> Running Command:
>>>     flatpak remote-add --user --if-not-exists flathub https://flathub.org/repo/flathub.flatpakrepo
>>> Return code: 0
Ensuring Flatpak runtime repo is registered... done

>>> Running Command:
>>>     flatpak install --assumeyes --user flathub org.gnome.Platform/x86_64/42 org.gnome.SDK/x86_64/42
Looking for matches…
Skipping: org.gnome.Platform/x86_64/42 is already installed
Similar refs found for ‘org.gnome.SDK’ in remote ‘flathub’ (user):

   1) runtime/org.gnome.Sdk.Compat.i386/x86_64/42
   2) runtime/org.gnome.Sdk.Compat.aarch64/x86_64/42
   3) runtime/org.gnome.Sdk/x86_64/42
   4) runtime/org.gnome.Sdk.Docs/x86_64/42

Which do you want to use (0 to abort)? [0-4]: 
```

Reply:
Ah - there's a typo in your config: `org.gnome.SDK` - > `org.gnome.Sdk`.

Reply:
:| i see. thanks. that fixed it.